slh-dsa/src/hashes/sha2.rs

```diff
@@ -117,7 +144,8 @@ where
             .chain_update(pk_seed)
             .chain_update(&zeroes)
             .chain_update(adrs.compressed());
-        m.iter().for_each(|x| sha.update(x.as_slice()));
+        m.iter()
+            .for_each(|x| Update::update(&mut sha, x.as_slice()));
```

Comment:
Hitting the same problem as usual. This time really unclear how to avoid it without loosing access to all the QoL methods of `Digest`.